packages/squiggle-lang/__tests__/Reducer/Reducer_Peggy/Reducer_Peggy_Parse_type_test.res

```diff
@@ -0,0 +1,79 @@
+open Jest
+open Reducer_Peggy_TestHelpers
+
+describe("Peggy parse type", () => {
+  describe("type of", () => {
+    testParse("p: number", "{(::$_typeOf_$ :p #number)}")
+  })
+  describe("type alias", () => {
+    testParse("type index=number", "{(::$_typeAlias_$ #index #number)}")
+  })
+  describe("type or", () => {
+    testParse(
+      "answer: number|string",
+      "{(::$_typeOf_$ :answer (::$_typeOr_$ (::$_constructArray_$ (#number #string))))}",
+    )
+  })
+  describe("type function", () => {
+    testParse(
+      "f: number=>number=>number",
+      "{(::$_typeOf_$ :f (::$_typeFunction_$ (::$_constructArray_$ (#number #number #number))))}",
+    )
+  })
+  describe("high priority modifier", () => {
+    testParse(
+      "answer: number<-min<-max(100)|string",
+      "{(::$_typeOf_$ :answer (::$_typeOr_$ (::$_constructArray_$ ((::$_typeModifier_max_$ (::$_typeModifier_min_$ #number) 100) #string))))}",
+    )
+    testParse(
+      "answer: number<-memberOf([1,3,5])",
+      "{(::$_typeOf_$ :answer (::$_typeModifier_memberOf_$ #number (::$_constructArray_$ (1 3 5))))}",
+    )
+  })
+  describe("low priority modifier", () => {
+    testParse(
+      "answer: number | string $ opaque",
```

Comment:
You might have mentioned this before; but why are ``opaque`` types useful at this stage? We don't use them much in Rescript, not sure if they're really needed now.

Reply:
They are not intended for the user. But they are in the syntax for future internal use. 
Because building the library will use actual type declarations and there will be a time that we want to make the type opaque. The motivation for that is that I do not want to have a forever growing EvValue collection. Rather looking for an ability to create opaque internal types if building blocks exist.

Reply:
Got it, thanks! Makes sense, I think. I'm looking forward to seeing how that's used in practice.